src/Common/tests/CoreFx.Private.TestUtilities/System/AssertExtensions.cs

```diff
@@ -337,5 +337,42 @@ public static void Equal<T>(HashSet<T> expected, HashSet<T> actual)
                 throw new XunitException($"Expected: {string.Join(", ", expected)}{Environment.NewLine}Actual: {string.Join(", ", actual)}");
             }
         }
+
+        public delegate void AssertThrowsActionReadOnly<T>(ReadOnlySpan<T> span);
+
+        public delegate void AssertThrowsAction<T>(Span<T> span);
+
+        // Cannot use standard Assert.Throws() when testing Span - Span and closures don't get along.
+        public static void AssertThrows<E, T>(ReadOnlySpan<T> span, AssertThrowsActionReadOnly<T> action) where E : Exception
+        {
+            try
+            {
+                action(span);
+                Assert.False(true, "Expected exception: " + typeof(E).GetType());
```

Comment:
This would be better if you throw an `XunitException` instead of doing Assert.False which will throw an XunitFalseException. It will have a cleaner message and stack. Look at other APIs in this class: https://github.com/dotnet/corefx/pull/39186/files#diff-d1da3333043f89e3df89346d157f4c92R337